### PR TITLE
ClearlyDefined scan storage improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt [hunk 1]
```diff
@@ -265,7 +265,7 @@ interface ClearlyDefinedService {
      * sense for the provider), see
      * https://api.clearlydefined.io/api-docs/#/harvest/get_harvest__type___provider___namespace___name___revision___tool___toolVersion_
      */
-    @GET("harvest/{coordinates}/{tool}/{toolVersion}?form=streamed")
+    @GET("harvest/{coordinates}/{tool}/{toolVersion}?form=raw")
```
Comment (fviernau) [inline clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt:268]: commit message: seem to cause -> seem**s** to cause

File: plugins/scanners/scancode/src/main/kotlin/ScanCodeResultModel.kt [hunk 2]
```diff
@@ -19,11 +19,14 @@
...
+import java.io.File
+
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
 import kotlinx.serialization.json.JsonArray
 import kotlinx.serialization.json.JsonElement
-import kotlinx.serialization.json.JsonTransformingSerializer
-import kotlinx.serialization.serializer
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.jsonPrimitive
```
Comment (fviernau) [inline plugins/scanners/scancode/src/main/kotlin/ScanCodeResultModel.kt:29]: commit-message: Please provide a rationale, why this is needed.

Comment (sschuberth) [reply plugins/scanners/scancode/src/main/kotlin/ScanCodeResultModel.kt:29]: The functionality is being used in the next commit, but I've added an explanation nonetheless.

File: scanner/src/main/kotlin/CommandLinePathScannerWrapper.kt [hunk 3]
```diff
@@ -43,6 +44,12 @@ abstract class CommandLinePathScannerWrapper(override val name: String) : PathSc
      */
     abstract fun runScanner(path: File, context: ScanContext): String
 
+    /**
+     * Create a [ScannerDetails] from parsing [result] in a scanner-specific way. This function can be used if a scan
```
Comment (mnonnenmacher) [inline scanner/src/main/kotlin/CommandLinePathScannerWrapper.kt:48]: Nit: "Create ~~a~~ [ScannerDetails]"

File: scanner/src/test/kotlin/storages/ClearlyDefinedStorageTest.kt [hunk 4]
```diff
@@ -388,7 +387,7 @@ private fun Result<List<ScanResult>>.shouldBeValid(block: (ScanResult.() -> Unit
 
         val scanResult = it.first()
         scanResult.summary.licenseFindings.find { finding ->
-            finding.location.path == TEST_PATH && "Apache-2.0" in finding.license.licenses()
+            finding.location.path == TEST_PATH && "MIT" in finding.license.licenses()
```
Comment (mnonnenmacher) [inline scanner/src/test/kotlin/storages/ClearlyDefinedStorageTest.kt:390]: Commit message should also explain that the test asset is for a different library to explain this license change.

File: clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt [hunk 5]
```diff
@@ -51,10 +51,10 @@ import retrofit2.http.Query
 interface ClearlyDefinedService {
     companion object {
         /**
-         * The maximum number of elements to request at once in a chunked request. This value was chosen more or less
-         * arbitrary to keep the size of responses reasonably small.
+         * The maximum number of elements to request at once in a chunked request. The value is based on feedback from
+         * the ClearlyDefined team.
          */
-        const val MAX_REQUEST_CHUNK_SIZE = 100
+        const val MAX_REQUEST_CHUNK_SIZE = 500
```
Comment (mnonnenmacher) [inline clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt:57]: Commit message: Why?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt (hunk 5 above)
Comment (sschuberth) [reply clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt:57]: I've added a rationale to the commit message.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**